src/support/ios.c

```diff
@@ -831,6 +831,66 @@ size_t ios_copyuntil(ios_t *to, ios_t *from, char delim)
     return total;
 }
 
+//Copy until '\r', '\n' or '\r\n'
+size_t ios_copyline(ios_t *to, ios_t *from, int chomp)
+{
+    size_t nchomp = 0;
+    size_t total = 0, avail = (size_t)(from->size - from->bpos);
+    size_t ntowrite;
+    while (!ios_eof(from)) {
+        if (avail == 0) {
+            avail = ios_readprep(from, LINE_CHUNK_SIZE);
+            if (avail == 0)
+                break;
+        }
+        size_t written;
+
+        char *r = NULL;
+        char *n = NULL;
+
+        for (size_t i = 0; i < avail; i++){
```

Comment:
> You have to look for both \r and \n in available bytes, but you only want to break on the first one found. e.g. If you have a large file and you look for \r both in avail (using memch) and never find it the code gets very slow.

I don't follow. What difference does it make from the manual loop? I just suggest breaking on the first match. Then you only have to call `strpbrk` again if there's a `\r` character without a following `\n`, which shouldn't be too frequent and isn't worth optimizing for.

> This has the same performance as readuntil which uses memchr.

Even, for relatively long strings? I'm asking because it looks like glibc contains hand-written assembly code using SSE instructions for this, so I figure there must be a reason.